test/components/enketo-iframe.spec.js

```diff
@@ -75,4 +76,27 @@ describe('EnketoIframe', () => {
 
     wrapper.vm.$router.push.calledWith('/projects/1').should.be.true;
   });
+
+  it('bubbles up the message event', async () => {
+    const wrapper = mountComponent({
+      props: { enketoId, actionType: 'new', instanceId: 'test-instance' },
+      container: {
+        router: mockRouter(`/?parentWindowOrigin=${window.location.origin}`)
+      }
+    });
+
+    const postMessage = sinon.fake();
+    sinon.replace(window.parent, 'postMessage', postMessage);
+
+    const iframe = wrapper.find('iframe');
+
+    const data = JSON.stringify({ enketoEvent: 'submissionsuccess' });
+    const script = document.createElement('script');
+    script.textContent = `window.parent.postMessage('${data}', "*");`;
+    iframe.element.contentDocument.body.appendChild(script);
+
+    await wait();
+
+    postMessage.calledWith(data, window.location.origin).should.be.true;
+  });
```

Comment:
In the component, `window.parent.postMessage()` is only called if `window.location !== window.parent.location`. Here we see that `postMessage()` ends being called, but I don't totally understand how the two locations differ in this test. I don't see either location being faked. Are the two locations different by default in testing?

Reply:
Because karma uses iframe to load test application the output of 

`console.log(window.location, window.parent.location);`
 is 
`http://localhost:9876/context.html, http://localhost:9876/?id=57567226`